v03_pipeline/lib/reference_datasets/misc.py

```diff
@@ -0,0 +1,31 @@
+import hail as hl
+
+from v03_pipeline.lib.model.dataset_type import DatasetType
+from v03_pipeline.lib.model.definitions import ReferenceGenome
+
+
+def enum_map(field: hl.Expression, enum_values: list[str]) -> dict:
+    lookup = hl.dict(
+        hl.enumerate(enum_values, index_first=False).extend(
+            # NB: adding missing values here allows us to
+            # hard fail if a mapped key is present and has an unexpected value
+            # but propagate missing values.
+            [(hl.missing(hl.tstr), hl.missing(hl.tint32))],
+        ),
+    )
+    if (
+        isinstance(field.dtype, hl.tarray | hl.tset)
+        and field.dtype.element_type == hl.tstr
+    ):
+        return field.map(
+            lambda x: lookup[x],
+        )
+    return lookup[field]
+
+
+def filter_contigs(ht, reference_genome: ReferenceGenome, dataset_type: DatasetType):
+    if dataset_type == DatasetType.MITO:
+        ht = ht.filter(ht.locus.contig == reference_genome.mito_contig)
```

Comment:
is it better to return ht here since contigs will already be filtered to mito?

Reply:
Yes!  Good suggestion!